Include/UrlLib/UrlLib.h

```diff
@@ -73,4 +81,36 @@ namespace UrlLib
 
         std::shared_ptr<Impl> m_impl{};
     };
+
+    class WebSocket final
+    {
+    public:
+        WebSocket();
+        ~WebSocket();
+
+        // Copy semantics
+        WebSocket(const WebSocket&);
+        WebSocket& operator=(const WebSocket&);
+
+        // Move semantics
+        WebSocket(WebSocket&&) noexcept;
+        WebSocket& operator=(WebSocket&&) noexcept;
+
+        ReadyState GetReadyState();
+        std::string GetURL();
+
+        void Open(std::string url,
+            std::function<void(void)> onopen,
+            std::function<void(void)> onclose,
+            std::function<void(std::string)> onmessage,
+            std::function<void(void)> onerror);
```

Comment:
The `onerror` should probably have a string associated for the error message?

Reply:
Discussed offline, to fix in another PR.